src/transactions.rs

```diff
@@ -0,0 +1,317 @@
+        });
+        serde_json::from_value(json)
+    }
+}
+
+pub type UpdateCredentialsPayloadDeserializeResult = DeserializeResult<UpdateCredentialsPayload>;
+
+/// Implements UDL definition of the same name.
+/// Deserialize [`UpdateCredentialsPayload`] from serialization format used by concordium nodes
+pub fn deserialize_update_credentials_payload(
+    bytes: Vec<u8>,
+) -> Result<DeserializeResult<UpdateCredentialsPayload>, ConcordiumWalletCryptoError> {
+    let fn_name = "deserialize_update_credentials_payload";
+
+    let mut bytes = std::io::Cursor::new(bytes);
+
+    let cred_infos_len: u8 = bytes.get().map_err(|e| e.to_call_failed(fn_name))?;
+    let cred_infos: AccountCredentialsMap =
+        deserial_map_no_length(&mut bytes, cred_infos_len.into())
+            .map_err(|e| e.to_call_failed(fn_name))?;
+
+    let remove_cred_ids_len: u8 = bytes.get().map_err(|e| e.to_call_failed(fn_name))?;
+    let remove_cred_ids = deserial_vector_no_length(&mut bytes, remove_cred_ids_len.into())
+        .map_err(|e| e.to_call_failed(fn_name))?;
+
+    let new_threshold =
+        AccountThreshold::deserial(&mut bytes).map_err(|e| e.to_call_failed(fn_name))?;
+
+    let payload = UpdateCredentialsPayload::try_from((cred_infos, remove_cred_ids, new_threshold))
+        .map_err(|e| e.to_call_failed(fn_name))?;
+
+    let result = DeserializeResult {
+        value: payload,
+        bytes_read: bytes.position(),
+    };
+    Ok(result)
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BakerKeysPayload {
+    /// New public key for participating in the election lottery.
+    pub election_verify_key: Bytes,
+    /// New public key for verifying this baker's signatures.
+    pub signature_verify_key: Bytes,
+    /// New public key for verifying this baker's signature on finalization
+    /// records.
+    pub aggregation_verify_key: Bytes,
+    /// Proof of knowledge of the secret key corresponding to the signature
+    /// verification key.
+    pub proof_sig: Bytes,
+    /// Proof of knowledge of the election secret key.
+    pub proof_election: Bytes,
+    /// Proof of knowledge of the secret key for signing finalization
+    /// records.
+    pub proof_aggregation: Bytes,
+}
+
+impl TryFrom<ConfigureBakerKeysPayload> for BakerKeysPayload {
+    type Error = serde_json::Error;
+
+    fn try_from(value: ConfigureBakerKeysPayload) -> Result<Self, Self::Error> {
+        serde_json::to_string(&value).and_then(|s| serde_json::from_str(&s))
+    }
+}
+
+pub fn make_configure_baker_keys_payload(
+    account_base58: String,
+    baker_keys: BakerKeyPairs,
+) -> Result<BakerKeysPayload, ConcordiumWalletCryptoError> {
+    let fn_desc = "make_configure_baker_keys_payload(...)";
+    let account = AccountAddress::from_str(&account_base58).map_err(|e| {
+        ConcordiumWalletCryptoError::CallFailed {
+            call: fn_desc.to_string(),
+            msg: format!("{:#}", e),
+        }
```

Comment:
```suggestion
    let account = AccountAddress::from_str(&account_base58)
    .map_err(|e| e.to_call_failed(fn_desc))?;
```

and then add in the `types.rs` file:

```
impl ConvertError for AccountAddressParseError {}
```

Reply:
Great suggestion! Thanks 😄